### ISSUE TITLE
Compiler.CompileFromFile not working in WinUI app

### ISSUE THREAD

**Issue by Gavin-Williams** [amerkoleci/Vortice.Windows#367]
When calling CompileFromFile (the shader compiler) in a WinUI app, the call fails with a file path not found error.

Here's a minimal WinUI app that shows the behavior.

[CompileShaderFromFile.zip](https://github.com/amerkoleci/Vortice.Windows/files/10080957/CompileShaderFromFile.zip)

**Comment by Gavin-Williams** [comment]
I can at least attest that the file can be fetched with the following Windows.Storage techniques...

`
// Use install directory to connect to file path
StorageFolder folderInstall = Package.Current.InstalledLocation;
StorageFolder folderAssets = await folderInstall.GetFolderAsync("Assets");
StorageFile file = await folderAssets.GetFileAsync("Color_VS.hlsl");
`
`
// Use app uri to connect to file path
StorageFile file = await StorageFile.GetFileFromApplicationUriAsync(new Uri("ms-appx:///Assets//Color_VS.hlsl"));
`

**Comment by amerkoleci** [comment]
Can't you just read the content and compile the string without calling CompileFromFilename?

**Comment by Gavin-Williams** [comment]
Yes I can, I'm also looking into the resource manager, see if it might be a good replacement for handling files the way I do. I guess I just like to know what's going on, it's usually my fault.

Could the function throw a NotImplementedException if that's the case? Or is the issue in DirectX?

**Comment by amerkoleci** [comment]
From msdn docs: https://learn.microsoft.com/en-us/windows/win32/api/d3dcompiler/nf-d3dcompiler-d3dcompilefromfile

` You can use this API to develop your Windows Store apps, but you can't use it in apps that you submit to the Windows Store. Refer to the section, "Compiling shaders for UWP", in the remarks for D3DCompile2.`
